### PR TITLE
Use a real nonempty list

### PR DESCRIPTION
**Pull request by NorfairKing**
This was prompted by the comment that said `-- non-empty list`.

### CODE REVIEW

File: subs/pantry/src/Pantry/Types.hs [hunk 1]
```diff
@@ -1272,7 +1274,8 @@ parseSnapshot mdir = withObjectWarnings "Snapshot" $ \o -> do
   snapshotHidden <- unCabalStringMap <$> (o ..:? "hidden" ..!= Map.empty)
   snapshotGhcOptions <- unCabalStringMap <$> (o ..:? "ghc-options" ..!= Map.empty)
   pure $ do
-    snapshotLocations <- fmap concat $ mapM (resolvePackageLocationImmutable mdir) unresolvedLocs
```
Comment (NorfairKing) [inline subs/pantry/src/Pantry/Types.hs]: This was `[NonEmpty a]`, so there is no way to concat to a `NonEmpty`

**Comment by dbaynard** [comment]
Hi @NorfairKing, I'll get to this in the next day or so.

**Comment by dbaynard** [comment]
…actually, I just checked out the travis failure. It seems there are more places in the code that are affected?

**Comment by snoyberg** [comment]
I ended up implementing this myself, forgetting that there was a PR for this already. Thanks for pointing it out @NorfairKing!

**Pull request closed without merging**